crates/telio-nurse/src/aggregator.rs

```diff
@@ -957,10 +1053,145 @@ mod tests {
         let segments = aggregator
             .collect_unacknowledged_segments(false)
             .await
+            .unwrap()
             .relay;
         assert_eq!(segments.len(), 0);
     }
 
+    #[tokio::test(start_paused = true)]
+    async fn test_aggregator_sleep_wakeup() {
+        let mut current_peer_event = create_basic_event(1);
+        let mut wg_interface = MockWireGuard::new();
+        let lambda_pk = current_peer_event.public_key;
+        wg_interface.expect_get_interface().returning(move || {
+            Ok(Interface {
+                peers: vec![(
+                    lambda_pk,
+                    Peer {
+                        public_key: lambda_pk,
+                        endpoint: None,
+                        rx_bytes: Some(2000),
+                        tx_bytes: Some(4000),
+                        ..Default::default()
+                    },
+                )]
+                .into_iter()
+                .collect(),
+                ..Default::default()
+            })
+        });
+        let mut aggregator = create_aggregator(false, true, wg_interface).await;
+
+        // Initial event
+        let first_segment_start = current_peer_event.timestamp;
+        aggregator
+            .change_peer_state_direct(
+                &current_peer_event,
+                EndpointProvider::Upnp,
+                EndpointProvider::Local,
+            )
+            .await;
+
+        current_peer_event.timestamp += Duration::from_secs(600);
+        let second_segment_start = current_peer_event.timestamp;
+        current_peer_event.rx_bytes += 1000;
+        current_peer_event.tx_bytes += 2000;
+        aggregator
+            .change_peer_state_direct(
+                &current_peer_event,
+                EndpointProvider::Upnp,
+                EndpointProvider::Upnp,
+            )
+            .await;
+
+        time::advance(Duration::from_secs(800)).await;
+
+        // Sleep is called
+        aggregator.collect_unacknowledged_segments(true).await;
+
+        let segments = aggregator
+            .collect_unacknowledged_segments(false)
```

Comment:
Mm. I think refactoring `collect_unacknowledged_segments` makes sense because from a third person perspective calling `collect_unacknowledged_segments(true)` seems like you are collecting the segments, and well vice versa.

I feel that moving the `force_close` and `24hr cycle` should be moved out of this function and this function just calls that. When we go to sleep, we just `save` the segments. I mean, right now by just reading the line it isn't 100% what will happen.

Reply:
Ok, so if you want to just save them, I guess the best option is to move it out to some segment-saving function (Probably some common private one and separate public wrappers for segment collection and storing them? Maybe it could take a predicate which events should be used for segment creation?)

Reply:
Yep. Thats what I am thinking

Reply:
Abstracted saving segments from collecting segments